### PR TITLE
Add fc-clip to the model zoo

### PR DESCRIPTION
**Pull request by neerajaabhyankar**
Adding another zero-shot segmentation model to the zoo, toward use in Auto Labelling.
Since this wasn't directly available on HF, used the open source repo https://github.com/bytedance/fc-clip/ to push a model and import from there.

```
Note: will move the model to the voxel51 HF org before merging.
```

## 📋 What changes are proposed in this pull request?

Added an integration for FC-CLIP to the model zoo.

## 🧪 How is this patch tested? If it is not, please explain why.

```
import fiftyone as fo
import fiftyone.zoo as foz

model = foz.load_zoo_model(
    "fc-clip-coco-torch",
    class_names=["cutsom", "class", "names"]
)

# load your `dataset`
dataset.apply_model(
    model,
    label_field="fcclip_pred",
)
```

### Is this a user-facing change that should be mentioned in the release notes?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release
      notes for FiftyOne users.

FC-Clip model for zero-shot instance segmentation added to the zoo.

### What areas of FiftyOne does this PR affect?

- [ ] App: FiftyOne application changes
- [ ] Build: Build and test infrastructure changes
- [ ] Core: Core `fiftyone` Python library changes
- [ ] Documentation: FiftyOne documentation changes
- [x] Other: Model zoo

## Summary by CodeRabbit

* **New Features**
  * Added support for FC-CLIP panoptic segmentation via a new model wrapper, including conversion of panoptic outputs into per-object detections with configurable class filtering and label assignment.
* **Style**
  * Improved readability of internal imports related to HuggingFace processor mappings.

---
**Enterprise sync PR**: https://github.com/voxel51/fiftyone-teams/pull/3277

### CODE REVIEW

File: fiftyone/utils/fcclip.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import fiftyone.utils.torch as fout
+
+
+class FCCLIPModelConfig(fout.TorchImageModelConfig, HasZooModel):
+    """Configuration for :class:`FCCLIPModel`.
+
+    Args:
+        name_or_path (None): HF repo or local path to the FC-CLIP model
+            uploaded with ``trust_remote_code=True``
+        score_threshold (0.8): minimum panoptic segment confidence to keep
```
Comment (manushreegangwar) [inline fiftyone/utils/fcclip.py]: Is this different from `confidence_thresh`?

File: fiftyone/utils/fcclip.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    """
+
+    def __init__(self, d):
+        d = self.init(d)
+        super().__init__(d)
+        self.name_or_path = self.parse_string(d, "name_or_path")
+        self.score_threshold = self.parse_number(
+            d, "score_threshold", default=0.8
+        )
+        self.class_names = self.parse_array(d, "class_names", default=None)
```
Comment (manushreegangwar) [inline fiftyone/utils/fcclip.py]: zero shot classes can directly override `classes` in config

File: fiftyone/utils/fcclip.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        model = foz.load_zoo_model("fc-clip-coco-torch")
+        dataset.apply_model(model, label_field="panoptic")
+        session = fo.launch_app(dataset)
+
+    Args:
+        config: an :class:`FCCLIPModelConfig`
+    """
+
+    def __init__(self, config):
+        self._hf_model = None
```
Comment (manushreegangwar) [inline fiftyone/utils/fcclip.py]: same question as other PR. why is this needed?

File: fiftyone/utils/fcclip.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            low_cpu_mem_usage=False,
+        )
+        self._hf_model.eval()
+        if config.class_names:
+            self._hf_model.set_class_names(config.class_names)
+
+        self._hf_model.config.object_mask_threshold = config.score_threshold
+        return self._hf_model
+
+    def _predict_all(self, imgs):
```
Comment (manushreegangwar) [inline fiftyone/utils/fcclip.py:243]: this should respect `TorchImageModel` format: preprocess is built into `transforms`, create batches using `collate_fn` and have a output processor for processing model outputs.

**Review by manushreegangwar — CHANGES_REQUESTED** [review]

File: fiftyone/utils/fcclip.py (hunk 3 above)
Comment (neerajaabhyankar) [reply fiftyone/utils/fcclip.py]: Just for readable declarations. Removed it for now since pylint didn't mind.

File: fiftyone/utils/fcclip.py (hunk 1 above)
Comment (neerajaabhyankar) [reply fiftyone/utils/fcclip.py]: seems the same -- renamed.

File: fiftyone/utils/fcclip.py (hunk 4 above)
Comment (neerajaabhyankar) [reply fiftyone/utils/fcclip.py:243]: Made a refactor similar to https://github.com/voxel51/fiftyone/pull/7912

File: fiftyone/utils/fcclip.py (hunk 2 above)
Comment (neerajaabhyankar) [reply fiftyone/utils/fcclip.py]: removed this override; now it's only set with config.classes

**Review by manushreegangwar — APPROVED** [review]

File: fiftyone/zoo/models/manifest-torch.json [hunk 5]
```diff
@@ -761,6 +761,7 @@
                 "type": "fiftyone.utils.fcclip.FCCLIPModel",
                 "config": {
                     "name_or_path": "Voxel51/fc-clip",
+                    "revision": "be25e3eac3dd8cc3c482b0b4251309bae3ab8419",
```
Comment (manushreegangwar) [inline fiftyone/zoo/models/manifest-torch.json:764]: Instead of  this, have you tried adding a tag: https://huggingface.co/docs/huggingface_hub/main/en/guides/repository#create-branches-and-tags?

Also, can you usee "version" for this purpose?

Comment (neerajaabhyankar) [reply fiftyone/zoo/models/manifest-torch.json:764]: Yeah tag looks cleaner.

```
{
            "base_name": "fc-clip-coco-instance-torch",
            "base_filename": null,
            "version": null,
            ...
            "default_deployment_config_dict": {
                "type": "fiftyone.utils.fcclip.FCCLIPModel",
                "config": {
                    "name_or_path": "Voxel51/fc-clip",
                    "revision": "be25e3eac3dd8cc3c482b0b4251309bae3ab8419",
                    "task": "instance"
                }
            },
            ...
```
you mean not the higher level "version" right? That could include the HF code + the FOE code

I'll rename revision --> tag
